src/Gastromio.Persistence.MongoDB/MongoDatabaseExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿using MongoDB.Bson;
+using MongoDB.Driver;
+using System.Threading.Tasks;
+
+namespace Gastromio.Persistence.MongoDB
+{
+    public static class MongoDatabaseExtensions
+    {
+        public static bool CollectionExists(this IMongoDatabase db, string collectionName)
+        {
+            var filter = new BsonDocument("name", collectionName);
+            //filter by collection name
+            var collections = db.ListCollections(new ListCollectionsOptions { Filter = filter });
+            //check for existence
+            return collections.Any();
+        }
+
+        public async static Task<bool> CollectionExistsAsync(this IMongoDatabase db, string collectionName)
```

Comment:
Erlaubt unten collections.AnyAsync die Übergabe eines CancellationTokens? Wenn ja, dann sollten wir das hier auch übergeben...

Reply:
CancellationTokens is now provided